### PR TITLE
MP-6060: Add customer specific prices documentation

### PR DESCRIPTION
**Pull request by ievgen-malykhin**
## PR Description

Add integration guide for customer specific prices feature

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+| PriceProductTableView | class | Created | src/Generated/Shared/Transfer/PriceProductTableViewTransfer |
+| PriceProductTableCriteria | class | Created | src/Generated/Shared/Transfer/PriceProductTableCriteriaTransfer |
+| PriceProductCriteria.withAllMerchantPrices | property | Created | src/Generated/Shared/Transfer/PriceProductCriteriaTransfer |
+| PriceProductDimension.idPriceProductDefault | property | Created | src/Generated/Shared/Transfer/PriceProductDimensionTransfer |
+| PriceProduct.priceTypeName | property | Created | src/Generated/Shared/Transfer/PriceProductTransfer |
+| PriceProductDimension.name | property | Created | src/Generated/Shared/Transfer/PriceProductDimensionTransfer |
+
+{% endinfo_block %}
+
+### 4) Configure price product group key pre-build plugins
```
Comment (sakharova-yuliia) [inline docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md:78]: All the plugins sections we can grouped in "Set up behavior" section with the plugins table accordingly.
Example: docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-order-management-feature-integration.md

File: docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+     * @return array<\Spryker\Zed\ProductMerchantPortalGuiExtension\Dependency\Plugin\PriceProductMapperPluginInterface>
+     */
+    protected function getPriceProductMapperPlugins(): array
+    {
+        return [
+            new MerchantRelationshipPriceProductMapperPlugin(),
+        ];
+    }
+}
+```
```
Comment (sakharova-yuliia) [inline docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md:198]: Please, add Verification block

File: docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+Make sure that the following changes were applied in transfer objects:
+
+| TRANSFER  | TYPE  | EVENT | PATH |
+| - | - | - | - |
+| PriceProductCollectionDeleteCriteria | class | Created | src/Generated/Shared/Transfer/PriceProductCollectionDeleteCriteriaTransfer |
+| PriceProductCollectionResponse | class | Created | src/Generated/Shared/Transfer/PriceProductCollectionResponseTransfer |
+| PriceProductMerchantRelationshipCollectionDeleteCriteria | class | Created | src/Generated/Shared/Transfer/PriceProductMerchantRelationshipCollectionDeleteCriteriaTransfer |
+| PriceProductMerchantRelationshipCollectionResponse | class | Created | src/Generated/Shared/Transfer/PriceProductMerchantRelationshipCollectionResponseTransfer |
+| PriceProductTableView | class | Created | src/Generated/Shared/Transfer/PriceProductTableViewTransfer |
```
[the remark this answers is not in the document]
Comment (ievgen-malykhin) [reply docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md:69]: It looks like nothing should be changed after Prerequisites installation.
Removed the chapter about transfer generation at all.

File: docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        ];
+    }
+}
+```
+
+</details>
+
+{% info_block warningBox "Verification" %}
+
+Log in to the Merchant Portal with a merchant with merchant relationships.
```
Comment (sakharova-yuliia) [inline docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md:189]:
```suggestion
Log in to the Merchant Portal with a merchant that has at least one merchant relationship. 
```

File: docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+Open any product for edit and make sure that the Customer column in the Price section is updatable.
```
Comment (sakharova-yuliia) [inline docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md:191]:
```suggestion
Open any merchant product and make sure that the Prices table containes "Customer" column. Make sure that you  can create a new customer price or even a butch of them by setting, amongst other things, an appropriate customer in the Customer field.  Make sure that you can delete the customer price. And make sure that you see the validation error while attempting to set or create the customer price for the volume price.```

Comment (sakharova-yuliia) [reply docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md:191]: As for me - I'd divide this verification into several blocks per plugin group installation to discribe what each group do.

File: docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+| --------- | ------ |-----------------|
+| Marketplace Merchant Custom Prices | {{page.version}} | |
+| Marketplace Merchant Portal Product Management | {{page.version}} | |
+
+spryker-feature/merchant-custom-prices
+spryker-feature/marketplace-merchant-portal-product-management
+
+### 1) Install the required modules using Composer
+
+Install the required module:
```
[the remark this answers is not in the document]
Comment (ievgen-malykhin) [reply docs/marketplace/dev/feature-integration-guides/202108.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md:28]: why module**s**, if only one module required?

**Review by sakharova-yuliia — DISMISSED** [review]

File: docs/marketplace/dev/feature-integration-guides/202200.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+| MODULE                                            | EXPECTED DIRECTORY        |
+|---------------------------------------------------|---------------------------|
+| PriceProduct                                      | vendor/spryker/price-product |
+| PriceProductMerchantRelationship                  | vendor/spryker/price-product-merchant-relationship |
+| PriceProductMerchantRelationshipMerchantPortalGui | vendor/spryker/price-product-merchant-relationship-merchant-portal-gui |
+| ProductMerchantPortalGui                          | vendor/spryker/product-merchant-portal-gui |
+
+{% endinfo_block %}
+
```
Comment (asmarovydlo) [inline docs/marketplace/dev/feature-integration-guides/202200.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md:43]: The transfer generation section is missing. Please do according to https://spryker.atlassian.net/wiki/spaces/DOCS/pages/478118714/Feature+integration+guides

File: docs/marketplace/dev/feature-integration-guides/202200.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+     */
+    protected function getPreBuildPriceProductGroupKeyPlugins(): array
+    {
+        return [
+            new MerchantRelationshipPreBuildPriceProductGroupKeyPlugin(),
+        ];
+    }
+}
+```
+
```
Comment (asmarovydlo) [inline docs/marketplace/dev/feature-integration-guides/202200.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md:150]: The verification step is missing.

Comment (asmarovydlo) [inline docs/marketplace/dev/feature-integration-guides/202200.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md:150]: The verification step is missing.

File: docs/marketplace/dev/feature-integration-guides/202200.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+            new MerchantRelationshipPriceProductMapperPlugin(),
+        ];
+    }
+}
+```
+</details>
+
+{% info_block warningBox "Verification" %}
+
+Log in to the Merchant Portal with a merchant that has at least one merchant relationship.
```
Comment (asmarovydlo) [inline docs/marketplace/dev/feature-integration-guides/202200.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md:119]: Please check all the verification steps and plugin specifications with Gramarly.

File: _data/sidebars/marketplace_dev_sidebar.yml [hunk 10]
```diff
@@ -429,6 +429,11 @@ entries:
             include_versions:
             - "202108.0"
 
+          - title: Merchant Portal - Marketplace Customer Specific Prices feature integration
```
Comment (asmarovydlo) [inline _data/sidebars/marketplace_dev_sidebar.yml:432]: Please use the correct feature name according to https://release.spryker.com/features/view/136 here & elsewhere. (Please do not forget to replace the links)

File: docs/marketplace/dev/feature-integration-guides/202200.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md (hunk 7 above)
Comment (ievgen-malykhin) [reply docs/marketplace/dev/feature-integration-guides/202200.0/merchant-portal-marketplace-customer-specific-prices-feature-integration.md]: The translation generation section is added as well.

**Review by asmarovydlo — APPROVED** [review]

File: docs/marketplace/dev/feature-integration-guides/202204.0/marketplace-merchant-custom-prices.md [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+}
+```
+
+</details>
+
+{% info_block warningBox "Verification" %}
+
+Open any merchant product with a regular price.
+Create a customer-specific price with the same combination of currency and country as the existing price.
+Make sure that there is no validation error.
```
Comment (AdvaBD) [inline docs/marketplace/dev/feature-integration-guides/202204.0/marketplace-merchant-custom-prices.md:185]: maybe also check the correct price on Yves?

File: docs/marketplace/dev/feature-integration-guides/202204.0/marketplace-merchant-custom-prices.md [hunk 12]
```diff
@@ -0,0 +1,232 @@
...
+    }     
+}
+```
+
+</details>
+
+{% info_block warningBox "Verification" %}
+
+Make sure that you see the validation error while attempting to set or create the customer price for the volume price.
+Make sure that you can delete the customer price.
```
Comment (AdvaBD) [inline docs/marketplace/dev/feature-integration-guides/202204.0/marketplace-merchant-custom-prices.md:230]: Make sure that you can edit and delete the customer price.

File: docs/marketplace/dev/feature-integration-guides/202204.0/marketplace-merchant-custom-prices.md [hunk 13]
```diff
@@ -0,0 +1,232 @@
...
+        ];
+    }     
+}
+```
+
+</details>
+
+{% info_block warningBox "Verification" %}
+
+Make sure that you see the validation error while attempting to set or create the customer price for the volume price.
```
Comment (AdvaBD) [inline docs/marketplace/dev/feature-integration-guides/202204.0/marketplace-merchant-custom-prices.md:229]: can we rephrase it to say customer-specific prices don't include/ support volume prices / cannot be combined with volume prices

**Review by AdvaBD — CHANGES_REQUESTED** [review]
